cmd/thv-operator/pkg/registryapi/podtemplatespec.go

```diff
@@ -196,17 +228,28 @@ func WithRegistrySourceMounts(containerName string, registries []mcpv1alpha1.MCP
 	}
 }
 
-// Apply applies the given options to the PodTemplateSpecBuilder.
-func (b *PodTemplateSpecBuilder) Apply(opts ...PodTemplateSpecOption) *PodTemplateSpecBuilder {
-	for _, opt := range opts {
-		opt(b.podTemplateSpec)
+// ParsePodTemplateSpec parses a runtime.RawExtension into a PodTemplateSpec.
+// Returns nil if the raw extension is nil or empty.
+// Returns an error if the raw extension contains invalid PodTemplateSpec data.
+func ParsePodTemplateSpec(raw *runtime.RawExtension) (*corev1.PodTemplateSpec, error) {
+	if raw == nil || raw.Raw == nil || len(raw.Raw) == 0 {
+		return nil, nil
 	}
-	return b
+
+	var pts corev1.PodTemplateSpec
```

Comment:
if we parse it as a `corev1.PodTemplateSpec`, why is the field defined as a `runtime.RawExtension` instead?
(just asking, I see it's consistent with the MCPServer type)